networkx/generators/nonisomorphic_trees.py

```diff
@@ -25,12 +25,48 @@ def nonisomorphic_trees(order):
 
     Yields
     ------
-    list of `networkx.Graph` instances
-       A list of nonisomorphic trees
-    """
+    `networkx.Graph` instances
+       A tree with `order` number of nodes that is not isomorphic to any other
+       yielded tree.
+
+    Raises
+    ------
+    ValueError
+        If `order` is negative
+
+    Examples
+    --------
+    There are 11 unique (non-isomorphic) trees with 7 nodes.
+
+    >>> n = 7
+    >>> nit_list = list(nx.nonisomorphic_trees(n))
+    >>> len(nit_list) == nx.number_of_nonisomorphic_trees(n) == 11
+    True
+
+    All trees yielded by the generator have the specified order.
+
+    >>> all(len(G) == n for G in nx.nonisomorphic_trees(n))
+    True
 
-    if order < 2:
-        raise ValueError
+    Each tree is nonisomorphic to every other tree yielded by the generator
+    >>> seen = []
+    >>> for G in nx.nonisomorphic_trees(n):
+    ...     assert not any(nx.is_isomorphic(G, H) for H in seen)
+    ...     seen.append(G)
+
+    See Also
+    --------
+    number_of_nonisomorphic_trees
+    """
+    if order < 0:
+        raise ValueError("order must be non-negative")
```

Comment:
A question (that's not strictly related to the proposal here, so feel free to disregard!).
This has come up in another discussion: if we were to standardize error messages, should they state what the problem is ("order is negative"), or should they tell the user what to do instead ("order must be non-negative")?

Reply:
Personally I have no opinion - my criteria are a) clarity and b) brevity (both subjective, I know). A "good" exception message makes it clear what the problem is as concisely as possible!

Warnings, IMV, are another matter - especially `FutureWarning` s and `DeprecationWarning` s. The best recipe for these IMO is to provide the new proposed pattern (e.g. `"foo(x) is deprecated, use bar(x) instead"`).